### ISSUE TITLE
Programs that resize themselves

### ISSUE THREAD

**Issue by woky** [i3/i3#2101]
Some programs wants to control their dimensions, for example Qmmp (something like WinAmp). Its player window can't be resized and making it non-floating results in this: https://i.imgur.com/lm2rxWa.png. That's bad example though as it's typical floating candidate. If you resize its playlist in classic window manager, the playlist window will resize by steps, which is somehow controlled by Qmmp. But i3, even when the playlist is floating, forces its dimensions which results in this https://i.imgur.com/xBXaDSi.png. There's also Tor Browser which forces its dimensions to best match user desired dimensions for privacy reasons and it results in this https://i.imgur.com/f7NvUGA.png or this https://i.imgur.com/MiaLocu.png.

Could i3 possibly repaint the background of the non-floating window of a program to some defined color (black) after it tries to resize itself?

Could i3 let floating windows resize themselves?

    $ i3 --version
    i3 version 4.10.4 (2015-09-08, branch "4.10.4") © 2009-2014 Michael Stapelberg and contributors

EDIT: Hmm, I was too fast. Floating Tor Browser can apparently resize itself. The repainting would still be nice.

**Comment by i3bot** [comment]
I don’t see a link to logs.i3wm.org. Did you follow http://i3wm.org/docs/debugging.html? (In case you actually provided a link to a logfile, please ignore me.)

**Comment by i3bot** [comment]
I don’t see a version number. Could you please copy & paste the output of `i3 --version` into this issue?

**Comment by Airblader** [comment]
> ts player window can't be resized and making it non-floating results in this: 

That's the application's fault. The entire point to a window manager is that it is free to decide what to do with any configuration request. It can accept it, deny it entirely or modify it to its own liking. The client must behave accordingly.

> If you resize its playlist in classic window manager, the playlist window will resize by steps, which is somehow controlled by Qmmp. But i3, even when the playlist is floating, forces its dimensions which results in this

i3 does support those "steps", so if it doesn't work for an application, it might set the hints incorrectly – for example, only after mapping the window. The only way we can figure out what's wrong for a specific application is if you provide a log file. Please read the comment about it posted by @i3bot.

> Could i3 possibly repaint the background of the non-floating window of a program to some defined color (black) after it tries to resize itself?

I think this is already supported with `client.background`. See http://i3wm.org/docs/userguide.html#_changing_colors

> Could i3 let floating windows resize themselves?

It already does.

**Comment by woky** [comment]
Sorry, I hastened too much. I realized that this probably isn't even problem with i3 and the issue is so minor that I don't have much interest in debugging it further. I'm attaching the logs nevertheless.

Here's the log after starting i3, starting the Tor Browser, and making a split: http://sprunge.us/VNaj

It seems `client.background` has no effect. Note the duplicated menu icon on the right side of Firefox https://i.imgur.com/Ajv4wBq.png. After switching to another workspace and back it is repainted.

Here's the log after starting i3, starting the Qmpp and resizing its playlist.
http://sprunge.us/geBV
